xmtp_id/src/associations/verified_signature.rs

```diff
@@ -100,6 +110,59 @@ impl VerifiedSignature {
         ))
     }
 
+    pub fn from_passkey<Text: AsRef<str>>(
+        signature_text: Text,
+        public_key: &[u8],
+        signature: &[u8],
+        authenticator_data: &[u8],
+        client_data_json: &[u8],
+    ) -> Result<Self, SignatureError> {
+        let client_data: ClientDataJson = serde_json::from_slice(client_data_json)
+            .map_err(|_| SignatureError::InvalidClientData)?;
+
+        let signature_text = BASE64_URL_SAFE_NO_PAD.encode(signature_text.as_ref());
+        if signature_text != client_data.challenge {
+            // Challenge needs to match signature text
+            return Err(SignatureError::InvalidClientData);
+        }
+
+        // 1. Parse the public key from raw bytes
+        let verifying_key = match VerifyingKey::from_sec1_bytes(public_key) {
+            Ok(key) => key,
+            Err(_err) => return Err(SignatureError::InvalidPublicKey),
+        };
+
+        // 2. Parse the signature
+        let signature = match Signature::from_der(signature) {
+            Ok(sig) => sig,
+            Err(_) => return Err(SignatureError::Invalid),
+        };
+
+        // 3. Hash the client data
+        let mut hasher = Sha256::new();
+        hasher.update(client_data_json);
+        let client_data_hash = hasher.finalize();
```

Comment:
We don't have some utility function to make this a one-liner?

Reply:
We have a function that will output a string https://github.com/xmtp/libxmtp/blob/04b6bb8197d05b079a26f79128fb0113bdc3cac3/xmtp_id/src/associations/member.rs#L387-L391

I can make one.